src/containers/LeftNav/index.js

```diff
@@ -98,23 +99,24 @@ export const LeftNav = (props) => {
     <NavWrapper>
       { navVisible && <React.Fragment>
         <Heading>CONTRACT</Heading>
-        <SubHeading onClick={() => setCurrentEditor('contract')}>Contract Template</SubHeading>
-        <SubHeading onClick={() => setCurrentEditor('contract')}>Contract Text</SubHeading>
+        <SubHeading onClick={() => setCurrentEditor(null, 'contract')}>Contract</SubHeading>
         <br />
         <Heading>CLAUSES</Heading>
-        {
-          clauseNodes.map((clauseNode) => {
+        { Object.keys(clauses).length
+          ? clauseNodes.map((clauseNode) => {
             const { clauseid, src } = clauseNode.data.attributes;
+            if (!clauses[clauseid]) return null;
+            const { clauseTemplateRef } = clauses[clauseid];
```

Comment:
The relationship between `clauseid` and `clauseTemplateRef` isn't immediately obvious. Might be worth a combination of documentation/different variable naming/or just a comment(?).

Reply:
`clauseid` is the unique identifier for a clause, which is an instance of a clause template. `clauseTemplateRef` points to the unique id of the clause template of which that clause is an instance. This will allow us in the future to be able to keep track of which clause instances need to be aware of changes to a specific clause template. I will add comments to the code, thanks!

Reply:
@jeromesimeon I've updated the PR and re-requested your review

Reply:
It actually does make sense! Thanks.